app/src/org/commcare/activities/FormEntryActivityUIController.java

```diff
@@ -737,6 +738,12 @@ protected void updateFormRelevancies() {
                 shouldRemoveFromView.add(i);
                 continue;
             }
+
+            //If there was change(in particular image-remove) in an image widget, then update the form
+            if (oldWidgets.get(i) instanceof ImageWidget) {
+                activity.updateFormMediaToDisk();
```

Comment:
Shouldn't we check the value of the old and new widgets in this case and make sure they haven't changed? I think we call updateFormRelevencies in a number of situations beyond when media is discarded, and it could cause performance problems if we needed to save the form to disk frequently.